karate-tests/src/test/java/patients/healthcareWorker/updatePatientDetails/replace.feature

```diff
@@ -63,4 +66,158 @@ Feature: Patch patient - Replace data
     Examples:
       | nhsNumber     | property      | options!                             |  
       | 5900059243    | gender        | ['male', 'female', 'unknown']        |
-      | 5900043320    | birthDate     | ["1985-10-26", "1955-11-05"]         |
\ No newline at end of file
+      | 5900043320    | birthDate     | ["1985-10-26", "1955-11-05"]         |
+
+    Scenario: Healthcare worker can add, update and remove patient's emergency contact details
+      * def nhsNumber = '5900079066'
+      * configure headers = call read('classpath:auth/auth-headers.js') 
+      * path 'Patient', nhsNumber
+      * method get
+      * status 200
+      * def originalVersion = parseInt(response.meta.versionId)
+
+    # add emergency contact details
+      * configure headers = call read('classpath:auth/auth-headers.js') 
+      * header Content-Type = "application/json-patch+json"
+      * header If-Match = karate.response.header('etag')
+      * path 'Patient', nhsNumber
+      * request read('classpath:patients/requestDetails/add/emergencyContact.json')
+      * method patch
+      * status 200
+      * def versionIdAftAdd = response.meta.versionId
+      * def contactDetails = response.contact
+      * match response contains {contact: '#notnull' }
+      * def relationshipDetails = contactDetails[0].relationship
+      * def contactId = contactDetails[0].id
+      * def indexToUpdate = response.contact.findIndex(x => x.id == contactId)
+      * def contactPathToUpdateAndRemove = "/contact/" + indexToUpdate
+
+      # update emergency contact
+      * configure headers = call read('classpath:auth/auth-headers.js') 
+      * header Content-Type = "application/json-patch+json"
+      * header If-Match = karate.response.header('etag')
+      * def newMobileNumber = faker.phoneNumber()
+      * path 'Patient', nhsNumber
+      * request 
+      """
+      {
+        "patches":[
+          {"op":"replace",
+          "path":"#(contactPathToUpdateAndRemove)",
+          "value":{
+            "id": "#(contactId)",
+            "relationship":"#(relationshipDetails)",
+            "telecom":[
+              {"system":"phone",
+              "value":"#(newMobileNumber)"}
+              ]}}]}
+       """       
+      * method patch
+      * status 200
+      * def updatedContact = response.contact.find(x => x.id == contactId)
+      * def telecomValue = updatedContact.telecom[0].value
+      * def versionIdAftUpdate = response.meta.versionId
+      * match parseInt(versionIdAftUpdate) == parseInt(versionIdAftAdd) + 1
```

Comment:
Have I understood this correctly? Why would the versionIdAftAdd be the same as versionIdAftUpdate after the update?